pyaedt/application/Analysis.py

```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
+        """
+        self.odesign.AnalyzeAll()
+        return True
+
+    @aedt_exception_handler
+    def list_of_variations(self, setupname=None, sweepname=None):
+        """Return list of active variation for input setup.
+
+        Parameters
+        ----------
+        setupname : str, optional
+            Setup name. If ``None`` nominal adaptive will be used.
+        sweepname : str, optional
+            Sweep name. If ``None`` nominal adaptive will be used.
+
+        Returns
+        -------
+        list
+        """
+
+        if not setupname and ":" in self.nominal_sweep:
+            setupname = self.nominal_adaptive.split(":")[0].strip()
+        elif not setupname:
+            self.logger.warning("No Setup defined.")
+            return False
+        if not sweepname and ":" in self.nominal_sweep:
+            sweepname = self.nominal_adaptive.split(":")[1].strip()
+        elif not sweepname:
+            self.logger.warning("No Setup defined.")
+            return False
+        if self.solution_type == "HFSS3DLayout" or self.solution_type == "HFSS 3D Layout Design":
+            try:
+                return list(self.osolution.ListVariations("{0} : {1}".format(setupname, sweepname)))
+            except:
+                return [""]
+        else:
+            try:
+                return list(self.odesign.ListVariations("{0} : {1}".format(setupname, sweepname)))
+            except:
+                return [""]
+
+    @aedt_exception_handler
+    def export_results(self, analyze=False, export_folder=None):
+        """Export all available reports to file, including sNp, profile and convergence.
+
+        Parameters
+        ----------
+        analyze : bool
+            Either to Analyze before export or not. Solutions have to be present for the design.
+        export_folder : str, optional
+            Full path to project folder.
+
+        Returns
+        -------
+        list
+            List of all exported files.
+        """
+        exported_files = []
+        if not export_folder:
+            export_folder = self.project_path
+        if analyze:
+            self.analyze_all()
+        setups = self.oanalysis.GetSetups()
+        if self.solution_type == "HFSS3DLayout" or self.solution_type == "HFSS 3D Layout Design":
+            excitations = len(self.oexcitation.GetAllPortsList())
+        else:
+            excitations = self.oboundary.GetNumExcitations()
+        reportnames = self.post.oreportsetup.GetAllReportNames()
+        for rName in reportnames:
+            rName2 = rName.replace(" ", "_")
+            self.post.oreportsetup.UpdateReports([str(rName)])
+            szExportPath = os.path.join(export_folder,
+                                        "{0}_{1}_{2}.csv".format(self.project_name, self.design_name, rName2))
+            self.post.oreportsetup.ExportToFile(str(rName), szExportPath)
+            self.logger.debug("Export Data: {}".format(szExportPath))
```

Comment:
@maxcapodi78 Do we want to log debug or info here?

Reply:
info